src/main/java/com/exasol/projectkeeper/repository/GitRepository.java

```diff
@@ -0,0 +1,151 @@
+package com.exasol.projectkeeper.repository;
+
+import static org.eclipse.jgit.lib.Constants.R_TAGS;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.*;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.*;
+import org.eclipse.jgit.revwalk.*;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This gives access to a local git repository.
+ */
+public class GitRepository {
+    private final Path projectDirectory;
+
+    /**
+     * Create a new instance of {@link GitRepository}.
+     * 
+     * @param projectDirectory the projects directory.
+     */
+    public GitRepository(final Path projectDirectory) {
+        this.projectDirectory = projectDirectory;
+    }
+
+    /**
+     * Get the tags of the current branch ordered descending by commit date. If one commit has multiple tags, they are
```

Comment:
This solution is required to also cover more complex repository structures.

For example, if we would have to maintain older versions of a project. So let's say we released 1.0 and then 2.0 and now need a fix for 1.0 since customers still need it. In that case, we would create a branch from 1.0 and release 1.1 here.  In that case, pk should list the differences to 1.0 and not to 2.0.